lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -606,8 +606,10 @@ def data_editor(
                 - Mixing data types within a column can make the column uneditable.
                 - Additionally, the following data types are not yet supported for editing:
                   complex, list, tuple, bytes, bytearray, memoryview, dict, set, frozenset,
-                  datetime.timedelta, decimal.Decimal, fractions.Fraction, pandas.Interval,
-                  pandas.Period, pandas.Timedelta.
+                  fractions.Fraction, pandas.Interval, and pandas.Period.
+                - To prevent overflow in JavaScript, datetime.timedelta and pandas.Timedelta
+                  columns will default to uneditable but this can be changed through column
```

Comment:
nit: maybe we can use something like `data types` here instead of `columns` to not mix it up too much with our definition of column types. Since we don't really have a Timedelta column, our NumberColumn is just able to handle timedelta values similar to how it also handles int, float, decimal types.

Reply:
Or maybe write something like: columns containing datetime.timedelta or pandas.Timedelta values ...

Reply:
Updated.